routers/web/shared/project/column.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package project
+
+import (
+	project_model "code.gitea.io/gitea/models/project"
+	"code.gitea.io/gitea/modules/json"
+	"code.gitea.io/gitea/services/context"
+)
+
+// MoveColumns moves or keeps columns in a project and sorts them inside that project
+func MoveColumns(ctx *context.Context) {
+	project, err := project_model.GetProjectByID(ctx, ctx.ParamsInt64(":id"))
+	if err != nil {
+		ctx.NotFoundOrServerError("GetProjectByID", project_model.IsErrProjectNotExist, err)
+		return
+	}
+	if project.OwnerID > 0 && project.OwnerID != ctx.ContextUser.ID {
```

Comment:
Why not check "can write project"?

Reply:
The write project permission has been checked on the `routers` definition functions.

Reply:
What does it do here?

Reply:
The write project permission check will only check the current login user has write permission to the repository's project unit. However it will not check whether the project belongs to this repository or this organization. This checks will avoid the request like `/<owner>/<repo>/projects/<non-current-repo-project-id>`

Reply:
Then it still needs to introduce a general function `can write to project {id}`? It seems that it is not the only call.

Reply:
But all of them are only 1 line, I don't think that's related to the bug fixes. Maybe other PRs.

Reply:
I do not see "1 line" is a reason. All other "can read" or "can write" are also all just 1 line.

----

Now, it is indeed 1 line, but they are all copied & pasted and there is no test. You could have 9 places right, but if one place goes wrong, then there is a security problem.